src/main/scala/analysis/GlobalRegionAnalysis.scala

```diff
@@ -0,0 +1,246 @@
+package analysis
+
+import analysis.solvers.SimpleWorklistFixpointSolver
+import ir.*
+
+import scala.collection.mutable
+
+trait GlobalRegionAnalysis(val program: Program,
+                           val domain: Set[CFGPosition],
+                           val constantProp: Map[CFGPosition, Map[Variable, FlatElement[BitVecLiteral]]],
+                           val reachingDefs: Map[CFGPosition, (Map[Variable, Set[Assign]], Map[Variable, Set[Assign]])],
+                           val mmm: MemoryModelMap,
+                           val globalOffsets: Map[BigInt, BigInt],
+                           val vsaResult: Option[Map[CFGPosition, LiftedElement[Map[Variable | MemoryRegion, Set[Value]]]]]) {
```

Comment:
it doesn't really make sense to have a Map inside an Option - you can just use an empty Map as the default instead of None.